### DISCUSSION TITLE
pep621: why does Renovate prepend the group name to the package name?

### DISCUSSION THREAD

**Discussion by maxbrunet** [renovatebot/renovate#27372] [Request Help] [answered]
### How are you running Renovate?

None

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

The manager prepend the group name to all dependencies in the `project.optional-dependencies` and `tool.pdm.dev-dependencies` sections:

https://github.com/renovatebot/renovate/blob/240d7b6a97b6e1a46a83172b79083f8a83b1ed99/lib/modules/manager/pep621/utils.ts#L92

For example:

```toml
[tool.pdm.dev-dependencies]
dev = ["pytest==8.0.0"]
```

Results in `dev/pytest` as the package name.

This adds extra works to the user to group their dependencies:

<img width="655" alt="image" src="https://github.com/renovatebot/renovate/assets/139807134/8e95b2d1-b6b8-4beb-a9da-58d1d8888e28">

Would it be something we could change? Either:

- remove the `groupName` prefix
- or an other option that would make more sense to me, suffix it to the `depType` (e.g. `dev-dependencies/dev`), although there are a few useful presets applying to the `dev-dependencies` depType, so it could be inconvenient as well

I would be happy to make the PR 🙂

### Logs (if relevant)

N/A

**Comment by viceice** [comment]
@secustor you wrote that code. is there a specific reason for the strange depName?🤗

- https://github.com/renovatebot/renovate/pull/22082

**Reply by secustor** [reply, marked as answer]
The assumption  at that time has been that users will want to have different PRs if they define this in multiple groups. 

Not sure where the second pytest package comes from in this example. 

Anyway there is somewhere a discussion already regarding this and IIRC we decided to change that in the next major.
